api/v1alpha1/nodemaintenance_types.go

```diff
@@ -21,11 +21,17 @@ import (
 )
 
 const (
-	// ConditionTypeReady is the Ready  Condition.Type for NodeMaintenance
+	// ConditionTypeReady is the Ready Condition type for NodeMaintenance.
+	// It is used to convey node readiness for maintenance operation by the requestor.
 	ConditionTypeReady string = "Ready"
+	// ConditionTypeFailed is the Failed Condition type for NodeMaintenance.
+	// It is used to convey failure during node maintenance operation by the requestor.
+	ConditionTypeFailed string = "Failed"
```

Comment:
we could use a more specific condition like "MaintenanceFailed". open to suggestions.
ATM chose the minimalistic approach.

later on we can consider adding the "Degraded" condition type to convey possible errors during Maintenance prep by the operator.

Reply:
the name is fine, let's discuss the reason why we need a separate condition